### ISSUE TITLE
Locking issue on sparc64 when opening files

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#886]
**Reported by misterx on 27 May 2011 04:04 UTC**

I acquired an UltraSparc IIIi system recently and have been unsuccessful getting aircrack-ng utility to work properly on both Debian Squeeze and Solaris Express 11. Both had some minor issues with stuff in src/osdep/, but which I was able to correct. However, both systems ended up with the same problem. Opening a pcap file with WPA2 seems to cause aircrack-ng to hang on a mutex lock attempt at aircrack-ng.c:1905.

There are lots of other pthreads apps that work fine on this platform, so there must be something else. But I'm not experienced enough to figure it out.

I tried ports that were done by other developers for Debian/Ubuntu that have the same problem. Opening a WPA cap file seems to work fine. The two test files included with aircrack-ng display this problem. wpa.cap works, wpa2.eapol.cap doesn't.

There is a aircrack-ng_1.0~rc1-2ubuntu1_sparc.deb that works just fine, however it seems not to use threads and is also much slower.

Does anyone have any ideas? Would be nice to get compatibility worked out for sparc. At least sparc/linux.

GDB output:

```
[debugging using libthread_db enabled](Thread)
[Thread 0xf7c1bb70 (LWP 14501)](New)
Opening /tmp/Dance-01.cap
^Cading packets, please wait...
Program received signal SIGINT, Interrupt.
0xf7fa379c in pthread_join () from /lib/ultra3/libpthread.so.0
(gdb) thread 2
[to thread 2 (Thread 0xf7c1bb70 (LWP 14501))](Switching)#0  0xf7fab67c in __lll_lock_wait () from /lib/ultra3/libpthread.so.0
(gdb) backtrace
#0  0xf7fab67c in __lll_lock_wait () from /lib/ultra3/libpthread.so.0
#1  0xf7fa5168 in pthread_mutex_lock () from /lib/ultra3/libpthread.so.0
#2  0x00017084 in check_thread (arg=0xffffdec9) at aircrack-ng.c:1905
#3  0xf7fa2458 in start_thread () from /lib/ultra3/libpthread.so.0
#4  0xf7d492dc in ?? () from /lib/ultra3/libc.so.6
#5  0xf7d492dc in ?? () from /lib/ultra3/libc.so.6
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) list 1905
1900            }
1901         }
1902   
1903         /* prevent concurrent access on the linked list */
1904   
1905         pthread_mutex_lock( &mx_apl );
1906   
1907         nb_pkt++;
1908   
1909         if( fmt == FORMAT_CAP )
```

See http://forum.aircrack-ng.org/index.php?topic=11227

**Comment by aircrack-ng** [comment]
**Comment by clopez on 27 May 2011 04:04 UTC**

Could you please try the package from Debian/sid and report if it works for you?

* http://packages.debian.org/sid/aircrack-ng


It should work on squeeze as is without problems (not need to recompile it I think)

**Comment by jbenden** [comment]
Closing as Solaris 11 patches landed in #2010 which show successful operation of unit-tests, and therefore, proper reading of PCAP files and threading of aircrack-ng application.
